### PR TITLE
[enforce.h Decouple logging.h] Delete glog/logging.h from enforce.h

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
Function optimization

### PR changes
Others

### Describe
Pcard-66989

After exposing the header files of phi, error will occur when users include header files which depend on "gflags.h" and "glog/logging.h", so that we need to decouple such dependence of header files.

Changes
This PR removes "glog/logging.h" of enforce.h.

### CODE REVIEW

File: paddle/phi/core/enforce.h
Comment (zyfncg) [inline paddle/phi/core/enforce.h:1]: image
290行有一处LOG使用，这里可能也需要处理下

Comment (jiahy0825) [reply paddle/phi/core/enforce.h:1]: 只有 `cuda_device_context_allocator.h` 文件用到这个宏，相应文件中已经做了处理～

Thanks～ In `cuda_device_context_allocator.h` file, `LOG` has already been handled.

**Review by zyfncg — DISMISSED** [review]

**Review by ZzSean — DISMISSED** [review]
LGTM for CI-OP-Benchmark

**Review by YuanRisheng — APPROVED** [review]

**Review by zh794390558 — APPROVED** [review]
LGTM

**Review by ZzSean — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**